src/System.Net.Http/src/System/Net/Http/Headers/KnownHeader.cs

```diff
@@ -9,42 +9,39 @@ namespace System.Net.Http.Headers
 {
     internal sealed class KnownHeader
     {
-        private readonly string _name;
-        private readonly HttpHeaderType _headerType;
-        private readonly HttpHeaderParser _parser;
-        private readonly string[] _knownValues;
-        private readonly byte[] _asciiBytesWithColonSpace;
-
-        public KnownHeader(string name) : this(name, HttpHeaderType.Custom, null)
+        public KnownHeader(string name, int? http2StaticTableIndex = null) : this(name, HttpHeaderType.Custom, null, null, http2StaticTableIndex)
         {
             Debug.Assert(!string.IsNullOrEmpty(name));
             Debug.Assert(HttpRuleParser.GetTokenLength(name, 0) == name.Length);
         }
 
-        public KnownHeader(string name, HttpHeaderType headerType, HttpHeaderParser parser, string[] knownValues = null)
+        public KnownHeader(string name, HttpHeaderType headerType, HttpHeaderParser parser, string[] knownValues = null, int? http2StaticTableIndex = null)
```

Comment:
Nit: It's far more common to have a http2StaticTableIndex than it is to have knownValues, so perhaps these should switch places in the arg list, so the definitions in KnownHeaders.cs are slightly more concise. Not a big deal.